### PR TITLE
Enable envd plugin for python 3.7

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
envd supports python 3.7, let's make the envd flytekit plugin in the 1.2 branch also support that

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 This is preventing us from generating the default docker images for the 1.2.12 [release](https://github.com/flyteorg/flytekit/actions/runs/5007729856/jobs/8975150027#step:7:668).

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: plugins/flytekit-envd/setup.py [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
+    python_requires=">=3.7",
     classifiers=[
         "Intended Audience :: Science/Research",
         "Intended Audience :: Developers",
         "License :: OSI Approved :: Apache Software License",
+        "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
-        "Programming Language :: Python :: 3.11",
```
Comment (eapolinario) [inline plugins/flytekit-envd/setup.py:29]: we do not support python 3.11 in the 1.2 branch.

**Review by pingsutw — APPROVED** [review]
